homeassistant/components/alexa/entities.py

```diff
@@ -614,10 +615,12 @@ def interfaces(self):
             yield AlexaChannelController(self.entity)
 
         if supported & media_player.const.SUPPORT_SELECT_SOUND_MODE:
-            inputs = AlexaInputController.get_valid_inputs(
+            inputs = AlexaEqualizerController.get_valid_inputs(
                 self.entity.attributes.get(media_player.const.ATTR_SOUND_MODE_LIST, [])
             )
-            if len(inputs) > 0:
+            if len(inputs) > 0 and self.entity_conf.get(CONF_PLATFORM) not in [
+                "denonavr"
```

Comment:
This works around the issue instead of addresses it. We shouldn't return data that breaks Alexa to begin with.

Reply:
I know, but I tried and I can't find a way to have denon/marantz receivers work with the EqualizerController. This causes the whole alexa platfrom to not be able to discover other devices, so it is a big problem impacting much more than only the denon/marantz receiver itself.
Besides I know this problem is experianced by multiple users: 5 in this issue https://github.com/home-assistant/core/issues/42294 and some more in this issue: https://github.com/home-assistant/core/issues/34029 that was already closed.

Those issue have been open since Oktober 2020, so I don't think there is currently anyone who knows how to properly fix this.

So my logical conclusion was: disable the EqualizerController for denon/marantz receivers so to make the discovery of alexa work at least to not impact other devices.
I could just remove the whole EqualizerController, but I don't know if it is working for some other people with diffrent integrations than denon/marantz, it might upset those users. (There actually are not that many integrations supporting sound modes).

Reply:
@balloob schould I remove the whole EqualizerController, or what is your suggestion?

Reply:
So it means we are generating invalid data for Alexa for Denon devices. 

https://github.com/home-assistant/core/blob/dev/homeassistant/components/alexa/capabilities.py#L1946

This function checks for valid data and so it should be updated.

What data does Alexa generate for Denon?

Reply:
```
2020-10-28 08:26:13 DEBUG (MainThread) [hass_nabucasa.iot] Received message:
{'handler': 'alexa',
 'msgid': '???',
 'payload': {'directive': {'header': {'messageId': '???',
                                      'name': 'Discover',
                                      'namespace': 'Alexa.Discovery',
                                      'payloadVersion': '3'},
                           'payload': {'scope': {'token': '???',
                                                 'type': 'BearerToken'}}}}}

2020-10-28 08:26:14 DEBUG (MainThread) [hass_nabucasa.iot] Publishing message:
{'msgid': '???',
 'payload': {'event': {'header': {'messageId': '???',
                                  'name': 'Discover.Response',
                                  'namespace': 'Alexa.Discovery',
                                  'payloadVersion': '3'},
                       'payload': {'endpoints': [{'capabilities': [{'interface': 'Alexa.SceneController',
                                                                    'supportsDeactivation': False,
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa',
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'}],
                                                  'cookie': {},
                                                  'description': '??? '
                                                                 'via Home '
                                                                 'Assistant',
                                                  'displayCategories': ['SCENE_TRIGGER'],
                                                  'endpointId': 'scene#???',
                                                  'friendlyName': '???',
                                                  'manufacturerName': 'Home '
                                                                      'Assistant'},
                                                 
------------------------------------------------------------------------------------
                                                 

                                                 {'capabilities': [{'interface': 'Alexa.PowerController',
                                                                    'properties': {'proactivelyReported': True,
                                                                                   'retrievable': True,
                                                                                   'supported': [{'name': 'powerState'}]},
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa.Speaker',
                                                                    'properties': {'proactivelyReported': True,
                                                                                   'retrievable': True,
                                                                                   'supported': [{'name': 'volume'},
                                                                                                 {'name': 'muted'}]},
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa.PlaybackController',
                                                                    'supportedOperations': ['Next',
                                                                                            'Pause',
                                                                                            'Play',
                                                                                            'Previous'],
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa.PlaybackStateReporter',
                                                                    'properties': {'proactivelyReported': True,
                                                                                   'retrievable': True,
                                                                                   'supported': [{'name': 'playbackState'}]},
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'inputs': [{'name': 'GAME'},
                                                                               {'name': 'MEDIA '
                                                                                        'PLAYER'},
                                                                               {'name': 'TUNER'}],
                                                                    'interface': 'Alexa.InputController',
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa.ChannelController',
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'configurations': {'modes': {'supported': [{'name': 'MUSIC'},
                                                                                                               {'name': 'MOVIE'}]}},
                                                                    'interface': 'Alexa.EqualizerController',
                                                                    'properties': {'proactivelyReported': False,
                                                                                   'retrievable': False,
                                                                                   'supported': [{'name': 'mode'}]},
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa.EndpointHealth',
                                                                    'properties': {'proactivelyReported': True,
                                                                                   'retrievable': True,
                                                                                   'supported': [{'name': 'connectivity'}]},
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa',
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'}],
                                                  'cookie': {},
                                                  'description': 'media_player.marantz '
                                                                 'via Home '
                                                                 'Assistant',
                                                  'displayCategories': ['TV'],
                                                  'endpointId': 'media_player#marantz',
                                                  'friendlyName': 'speaker',
                                                  'manufacturerName': 'Home '
                                                                      'Assistant'},
                                                                      
                                                                      
------------------------------------------------------------------------------------
                                                                      
                                                 
                                                 {'capabilities': [{'interface': 'Alexa.PowerController',
                                                                    'properties': {'proactivelyReported': True,
                                                                                   'retrievable': True,
                                                                                   'supported': [{'name': 'powerState'}]},
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa.EndpointHealth',
                                                                    'properties': {'proactivelyReported': True,
                                                                                   'retrievable': True,
                                                                                   'supported': [{'name': 'connectivity'}]},
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'},
                                                                   {'interface': 'Alexa',
                                                                    'type': 'AlexaInterface',
                                                                    'version': '3'}],
                                                  'cookie': {},
                                                  'description': 'switch.??? '
                                                                 'via Home '
                                                                 'Assistant',
                                                  'displayCategories': ['SWITCH'],
                                                  'endpointId': 'switch#???',
                                                  'friendlyName': '??? '
                                                                  '??? '
                                                                  '???',
                                                  'manufacturerName': 'Home '
                                                                      'Assistant'}]}}}}
```

Reply:
The debug log is from my setup from the linked issue: https://github.com/home-assistant/core/issues/42294#issuecomment-717761785, it was a few HomeAssistant releases back, but the issue is still the same.
@balloob you thought then that is was this piece that was giving trouble: https://github.com/home-assistant/core/issues/42294#issuecomment-717786511
```
{
    "configurations": {"modes": {"supported": [{"name": "MUSIC"}, {"name": "MOVIE"}]}},
    "interface": "Alexa.EqualizerController",
    "properties": {
        "proactivelyReported": False,
        "retrievable": False,
        "supported": [{"name": "mode"}],
    },
}
```

If you want I can run the debug log again tonight and get you a fresch one on the latest HomeAssistant release (although I am pretty sure it will be the same).

Reply:
Can you try again with both #50724 and #50730 merged.

Reply:
@balloob, I tried with both #50724 and #50730 but it still does not work, the relevant debug message now changed to:

```
 {
    "configurations":{
       "modes":{
          "supported":[
             {
                "name":"MUSIC"
             },
             {
                "name":"MOVIE"
             }
          ]
       }
    },
    "interface":"Alexa.EqualizerController",
    "properties":{
       "proactivelyReported":false,
       "retrievable":true,
       "supported":[
          {
             "name":"mode"
          }
       ]
    },
    "type":"AlexaInterface",
    "version":"3"
 },
```

So the `"retrieveble"` changed to `True` as expected

Reply:
Meh. Okay. 

This fix is incorrect, please update the `if` check to check agains denon.

Reply:
@balloob do you know how to get the platform/integration from the state object so I can filter out the denonavr platform?

I can't seem to figure out how to properly check against denon in the if check...

Reply:
```python
from homeassistant.helpers.entity import entity_sources

domain = entity_sources(hass)[entity_id]['domain']
```

Reply:
Also include a comment why it's disabled.

Reply:
Thanks, testing now.